src/charm.py

```diff
@@ -554,6 +643,132 @@ def _update(self, event):
 
         self.unit.status = MaintenanceStatus("replanning application")
 
+    # Helpers for frontend TLS
+    def _relation_created(self, relation_name: str) -> bool:
+        return bool(self.model.relations.get(relation_name))
+
+    def _certificate_is_available(self) -> bool:
+        cert, key = self.certificates.get_assigned_certificate(
+            certificate_request=self._get_certificate_request_attributes()
+        )
+        return bool(cert and key)
+
+    def _get_certificate_request_attributes(self) -> CertificateRequestAttributes:
+        """Return the attributes of the certificate this charm will request."""
+        # Generate SANS IP - use the unit ip in case any clients try to reach
+        # the frontend server using it
+        unit_hostname = socket.getfqdn()
+        sans_ip = frozenset((socket.gethostbyname(unit_hostname),))
```

Comment:
Some TLS providers will refuse to sign a certificate with sans IPS (ex. Let's Encrypt). This may not be an issue on its own, but let's make sure we really need those sans_ip's in the cert. If we don't really need them I would avoid using them altogether.

Reply:
I have added configuration options for both SANS DNS and CN - in my case, removing the IPs means that I no longer can access the server using the unit IP, but I have to always have some DNS resolution (I just changed my /etc/hosts).

Please let me know if this makes sense.